webapp/client/client.js

```diff
@@ -5,20 +5,41 @@ export default class Client {
         this.url = '/plugins/github/api/v1';
     }
 
-    requestReviewers = async (prId, reviewers, org, repo) => {
-        return this.doPost(`${this.url}/pr/reviewers`, {pull_request_id: prId, reviewers, org, repo});
+    getConnected = async (reminder = false) => {
+        return this.doGet(`${this.url}/connected?reminder=` + reminder);
     }
 
-    removeReviewers = async (prId, reviewers, org, repo) => {
-        return this.doDelete(`${this.url}/pr/reviewers`, {pull_request_id: prId, reviewers, org, repo});
+    getReviews = async () => {
+        return this.doGet(`${this.url}/reviews`);
     }
 
-    editPr = async (prId, labels, assignees, milestone, org, repo) => {
-        return this.doPost(`${this.url}/pr`, {pull_request_id: prId, labels, assignees, milestone, org, repo});
+    getMentions = async () => {
+        return this.doGet(`${this.url}/mentions`);
+    }
+
+    getUnreads = async () => {
+        return this.doGet(`${this.url}/unreads`);
+    }
+
+    doGet = async (url, body, headers = {}) => {
+        headers['X-Requested-With'] = 'XMLHttpRequest';
```

Comment:
What purpose does `X-Requested-With` have?

Reply:
Great question, ongoing thread specifically discussing that here https://pre-release.mattermost.com/core/pl/3wbotmckrf87mekbuse4wun9sc